implementations/java-json-schema/BowtieJsonSchema.java

```diff
@@ -115,18 +120,42 @@ private void dialect(JsonNode node) throws JsonProcessingException {
       DialectRequest.class
     );
 
-    // well, setting default dialect shouldn't be necessary but why not
-    if (SpecificationVersion.DRAFT2020_12.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2020Dialect());
-    } else if (SpecificationVersion.DRAFT2019_09.getId().equals(dialectRequest.dialect())) {
-        validatorFactory.withDefaultDialect(new Dialects.Draft2019Dialect());
-    } else if (SpecificationVersion.DRAFT7.getId().equals(dialectRequest.dialect())) {
-      validatorFactory.withDefaultDialect(new Dialects.Draft7Dialect());
+    ConcurrentHashMap<String, Dialect> dialectsMap = Arrays.stream(Dialects.class.getClasses())
+            .filter(Dialect.class::isAssignableFrom)
+            .map(clazz -> {
+                try {
+                    return (Dialect) clazz.getConstructor().newInstance();
+                } catch (Exception e) {
+                    throw new IllegalStateException("Failed to instantiate Dialect", e);
+                }
+            })
+            .collect(
+              Collectors.toConcurrentMap(
```

Comment:
`ConcurrentHashMap` is not really necessary here:
- the whole harness code is sequential and single-threaded
- this is just a local variable in a method so even if there were multiple threads they'd cannot possibly share this map instance (each thread would create its own map instance)
- this map is created once and never changed, so it's effectively immutable. And immutable maps are thread-safe

And the other thing is that we can instantiate this map only once per harness lifetime. So I'd suggest making this map a final field in `BowtieJsonSchema` class and instantiate it in constructor (line 46). After this change it still wouldn't be necessary to change it to `ConcurrentHashMap`. But you can call `Collections.unmodifiableMap(...)` on a final result to be extra safe and actually enforce immutability :)

Reply:
https://results.pre-commit.ci/run/github/531839193/1724347296.qqr9ylU_Tz-p-HGmDO30nQ

Reply:
Ah, so that's why you changed it. So I would just disable this rule in PMD because it doesn't make sense (this tool was really noisy from the beginning, reports a lot of false positives). I don't remember where the config was located though and cannot check it right now

Reply:
> And the other thing is that we can instantiate this map only once per harness lifetime. So I'd suggest making this map a final field in BowtieJsonSchema class and instantiate it in constructor (line 46). After this change it still wouldn't be necessary to change it to ConcurrentHashMap. But you can call Collections.unmodifiableMap(...) on a final result to be extra safe and actually enforce immutability :)

The pre-commit error just went away after doing this :). Thanks!